### ISSUE TITLE
Utf8JsonWriterTests.WriteLargeKeyOrValue tests are failing

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#39856]
```
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: True, skipValidation: True) [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: True, skipValidation: False) [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: False, skipValidation: True) [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
        Assert.Equal() Failure
      System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(formatted: False, skipValidation: False) [FAIL]
        Expected: 0
        Actual:   1
        Stack Trace:
          d:\repos\corefx\src\System.Text.Json\tests\Utf8JsonWriterTests.cs(4996,0): at System.Text.Json.Tests.Utf8JsonWriterTests.WriteLargeKeyOrValue(Boolean formatted, Boolean skipValidation)
```
cc: @ahsonkhan

**Comment by stephentoub** [comment]
This was just broken by https://github.com/dotnet/corefx/commit/74d26be704117139e9ef081c9e287740fde5d002#diff-ad2e0c88c19fbbbf96791f5d54976a1bR4986.

**Comment by karelz** [comment]
Fixed in master (5.0) in PR #39857 and also in release/3.0 branch (3.0) in PR #39867.
